packages/cardhost/app/components/mode-switcher.js

```diff
@@ -0,0 +1,42 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+const modes = [
+  {
+    id: 'view',
+    displayTitle: 'Layout View'
+  },
+  {
+    id: 'edit',
+    displayTitle: 'Edit View'
+  },
+  {
+    id: 'schema',
+    displayTitle: 'Schema View'
+  }
+];
+
+export default class ModeSwitcher extends Component {
+  modes = modes;
+
+  @service router;
+  @tracked currentCardName;
+  @tracked currentMode;
+
+  constructor(...args) {
+    super(...args);
+
+    this.currentCardName = this.args.name;
+    this.currentMode = this.args.mode;
```

Comment:
There is no need to create this property--just use the arg you passed in. you only need to create component properties if it's a value that you need to mutate.